packages/view/src/Charcoal/View/Twig/TwigEngine.php

```diff
@@ -173,9 +183,17 @@ public function render(string $templateIdent, $context): string
      */
     public function renderTemplate(string $templateString, $context): string
     {
-        $template = $this->twig()->createTemplate($templateString);
-        $arrayContext = json_decode(json_encode($context), true);
-        return $template->render($arrayContext);
+        try {
+            $template = $this->twig()->createTemplate($templateString);
+            $arrayContext = json_decode(json_encode($context, JSON_THROW_ON_ERROR), true);
+            return $template->render($arrayContext);
+        } catch (JsonException $e) {
+            throw new Exception(
+                sprintf("Twig cannot render template \"%s\", Error : \"%s\".", $templateString, $e->getMessage()),
+                $e->getCode(),
+                $e
+            );
```

Comment:
```suggestion
            if (strlen($templateString) > 30) {
                // Truncate the string to avoid polluting the logs with a long template.
                $templateString = substr($templateString, 0, 29) . '…';
            }

            throw new UnexpectedValueException(
                sprintf('Twig cannot render template [%s]: %s', $templateString, $e->getMessage()),
                $e->getCode(),
                $e
            );
```